plugins/ingestion-kinesis/src/main/java/org/opensearch/plugin/kinesis/KinesisShardConsumer.java

```diff
@@ -247,6 +247,15 @@ public int getShardId() {
         return shardId;
     }
 
+    /**
+     * Kinesis does not support pointer based lag. Return default value of 0.
+     * @param expectedStartPointer the pointer where ingestion would start if no messages have been consumed yet
+     */
+    @Override
+    public long getPointerBasedLag(IngestionShardPointer expectedStartPointer) {
+        return 0;
+    }
```

Comment:
You are correct here. In particular, since Kinesis sequence numbers are not dense (and probably have a timestamp component to them), they're not useful for seeing how far behind you are.

The best we're likely to be able to do is use the `MillisBehindLatest` property in the `GetRecords` response.

I just read this blog post, which I found informative: https://www.waitingforcode.com/data-engineering-aws/kinesis-sequence-number-not-apache-kafka-offset/read